pkg/quota/evaluator/core/pods.go

```diff
@@ -144,92 +144,137 @@ func enforcePodContainerConstraints(container *v1.Container, requiredSet, missin
 }
 
 // podUsageHelper can summarize the pod quota usage based on requests and limits
-func podUsageHelper(requests v1.ResourceList, limits v1.ResourceList) v1.ResourceList {
-	result := v1.ResourceList{}
-	result[v1.ResourcePods] = resource.MustParse("1")
-	if request, found := requests[v1.ResourceCPU]; found {
-		result[v1.ResourceCPU] = request
-		result[v1.ResourceRequestsCPU] = request
+func podUsageHelper(requests api.ResourceList, limits api.ResourceList) api.ResourceList {
+	result := api.ResourceList{}
+	result[api.ResourcePods] = resource.MustParse("1")
+	if request, found := requests[api.ResourceCPU]; found {
+		result[api.ResourceCPU] = request
+		result[api.ResourceRequestsCPU] = request
 	}
-	if limit, found := limits[v1.ResourceCPU]; found {
-		result[v1.ResourceLimitsCPU] = limit
+	if limit, found := limits[api.ResourceCPU]; found {
+		result[api.ResourceLimitsCPU] = limit
 	}
-	if request, found := requests[v1.ResourceMemory]; found {
-		result[v1.ResourceMemory] = request
-		result[v1.ResourceRequestsMemory] = request
+	if request, found := requests[api.ResourceMemory]; found {
+		result[api.ResourceMemory] = request
+		result[api.ResourceRequestsMemory] = request
 	}
-	if limit, found := limits[v1.ResourceMemory]; found {
-		result[v1.ResourceLimitsMemory] = limit
+	if limit, found := limits[api.ResourceMemory]; found {
+		result[api.ResourceLimitsMemory] = limit
 	}
 	return result
 }
 
 // PodUsageFunc knows how to measure usage associated with pods
-func PodUsageFunc(object runtime.Object) v1.ResourceList {
-	pod, ok := object.(*v1.Pod)
-	if !ok {
-		return v1.ResourceList{}
-	}
-
+func PodUsageFunc(obj runtime.Object) api.ResourceList {
```

Comment:
Actually we need the conversion the other way around (v1 to api), because all the ResourceList logic is defined on api.ResourceList. So it's the efficiency of the resoucequota controller that gets affected. Is that acceptable?